src/lib/services/DeviceModel.ts

```diff
@@ -0,0 +1,6 @@
+export interface AirnoteDevice {
+	deviceUID: string;
+	productUID: string | (string | null)[];
+	pin: string | (string | null)[];
+	internalNav?: string | (string | null)[] | null;
```

Comment:
As a suggestion, maybe we could make some sort of custom type for `string | (string | null)[] | null` because you have to type that out a whole lot.

Reply:
Very good point. Hopefully you like `PotentiallyNullDeviceDetails`?

Reply:
Sure. Way better than `string | (string | null)[] | null` :)